### PR TITLE
doc: format the DataStats model [TCTC-1417]

### PR DESCRIPTION
**Pull request by Sanix-Darker**
# What
- Add and clarify the doc for the DataStats class

# Why
More details from this card : [TCTC-1417](https://toucantoco.atlassian.net/jira/software/c/projects/TCTC/boards/18?modal=detail&selectedIssue=TCTC-1417)

### CODE REVIEW

File: doc/DataSlice.md
Comment (davinov) [inline doc/DataSlice.md:20]: Maybe we should define what a query is in this context? It's important to know what this duration will mean

File: doc/DataSlice.md
Comment (davinov) [inline doc/DataSlice.md:55]: Instead of having psuedo code related to snowflake, maybe we could have a quick diagram to present the steps and their measure? Or just a list like:
```
Steps what happens in get_slice:
- A: modify query to get just the slice
- B: request the result of the modified query to the external service 
- C: turn them into a DataFrame
- D: filter them with permissions
- E: compute statistics and return

`execution time`: start of A -> end of D
`extraction time`: start of B -> end of B
`conversion time`: start of C -> end of C
```

What do you think of this way of explaining?

Comment (davinov) [reply doc/DataSlice.md:55]: Listing all the steps involved will also help structuring the timings we need to measure, and how to name them :)

File: doc/DataSlice.md
Comment (raphaelvignes) [inline doc/DataSlice.md:23]:
```suggestion
    * `permission_application_time` (`float`): in seconds, the time to filter data with permissions.
```

File: doc/DataSlice.md
Comment (raphaelvignes) [inline doc/DataSlice.md:27]: Est-ce qu'on ne doit pas enlever la mention à Snowflake ici ?

File: toucan_connectors/snowflake_common.py
Comment (raphaelvignes) [inline toucan_connectors/snowflake_common.py:68]:
```suggestion
        self.permission_application_time: Optional[float] = None
```

File: toucan_connectors/toucan_connector.py
Comment (raphaelvignes) [inline toucan_connectors/toucan_connector.py:38]:
```suggestion
    permission_application_time: Optional[float] = None
```

**Review by raphaelvignes — CHANGES_REQUESTED** [review]
Merci pour cette clarification ! 
Quelques remarques, si la demande de modif sur le nom `data_filtered_from_permission_time` implique des changements ailleurs que dans ce repo, pas obligé de prendre en compte.

File: doc/DataSlice.md
Comment (Sanix-Darker) [reply doc/DataSlice.md]: Logiquement oui, je penses, je viens d'update !

**Review by raphaelvignes — APPROVED** [review]
Thx !

**Comment by raphaelvignes** [comment]
Wait for @davinov review's before merging plz

File: doc/DataSlice.md
Comment (davinov) [inline doc/DataSlice.md:24]: In the model, it's called `compute_stats_time` https://github.com/ToucanToco/toucan-connectors/pull/546/files#diff-5feb095e781094a6657a509f814113ef04d1ecfefd4209765d89394564f57737R37
Which one do you want to keep?

(It's also `stats_computed_time` in the data service so I recommend keeping this one)

File: doc/DataSlice.md
Comment (davinov) [inline doc/DataSlice.md:23]: In the http data service, it's called `data_filtered_from_permission_time`. Which one do you want to keep? (the important thing is that it's the same everywhere).

File: toucan_connectors/snowflake_common.py
Comment (davinov) [inline toucan_connectors/snowflake_common.py:117]: Not necessarily in the PR but: what is the interest of having setters for these properties? It seems exactly the same than `self.query_generation_time = ` to me

**Review by davinov — COMMENTED** [review]
Still 2 naming coherence problems, and we'll be good to merge!
When we make a release for this, I recommend we write a changelog note about the changes, and bump the major version (as consumers like Toucan's Laputa are supposed to be impacted)

File: doc/DataSlice.md
Comment (raphaelvignes) [reply doc/DataSlice.md:23]: @Sanix-Darker If you prefer, you can revert my proposal.

**Comment by Sanix-Darker** [comment]
For the CHANGELOG updates, i think it will be better to add that on a bump PR... what do you think @davinov ?

File: toucan_connectors/snowflake_common.py
Comment (Sanix-Darker) [reply toucan_connectors/snowflake_common.py:117]: True, i just changed the names, because it was already there..
and am not sure, but... i think... 🤔 we don't know if this function get used by laputa... if we are only on the context of this PR... and some of the setters are used by other methods as callbacks... see [line 187](https://github.com/ToucanToco/toucan-connectors/blob/2e853eb1869db8746816ca54dda028c5ccb6d8e5/toucan_connectors/snowflake_common.py#L187)

The code will be adapt in consequence on laputa.

File: doc/DataSlice.md
Comment (Sanix-Darker) [reply doc/DataSlice.md]: 👍🏾

File: doc/DataSlice.md
Comment (Sanix-Darker) [reply doc/DataSlice.md]: 👍🏾

**Review by davinov — APPROVED** [review]
:+1: thx!

File: toucan_connectors/snowflake_common.py
Comment (davinov) [reply toucan_connectors/snowflake_common.py:117]: Yes, it's not related to this PR anyway ^^

**Pull request merged**
